File: arrays-101/valid-mountain.scala
object Solution {
    def validMountainArray(arr: Array[Int]): Boolean = {
        if (arr.length < 3) {
            return false
        }
        var (left, right) = arr.splitAt(arr.indexOf(arr.max))
        left = left :+ right(0)
        if (left.length == arr.length || right.length == arr.length) {
            return false
        }
        (0 until left.length - 1) foreach { i =>
            if (left(i) >= left(i + 1)) return false
        }
        (0 until right.length - 1) foreach { i =>
            if (right(i) <= right(i + 1)) return false
        }
        true
    }
}

File: arrays-101/valid-mountain.cpp
#include "stdio.h"
using namespace std;

class Solution {
public:
    bool validMountainArray(vector<int>& arr) {
        if (arr.size() < 3) {
            return false;
        }
        if (arr[0] >= arr[1] || arr[arr.size() - 2] <= arr[arr.size() - 1]) {
            return false;
        }
        bool increase = true;
        for (int i = 0; i < arr.size() - 1; i++)
        {
            if (increase) {
                if (arr[i] == arr[i+1]) {
                    return false;
                }
                if (arr[i] > arr[i+1]) {
                    increase = false;
                }
            } else {
                if (arr[i] <= arr[i+1]) {
                    return false;
                }
            }
        }
        return true;
    }
};